code/datums/abilities/flock/flockmind.dm

```diff
@@ -390,21 +388,33 @@
 	if(locate(/obj/flock_structure) in T)
 		boutput(holder.owner, "<span class='alert'>There is already a flock structure on this flocktile!</span>")
 		return 1
+
 	for (var/atom/O in T.contents)
 		if (O.density && !isflock(O))
 			boutput(holder.owner, "<span class='alert'>That tile has something that blocks tealprint creation!</span>")
 			return 1
+
+	var/list/friendlyNames = list()
+	var/mob/living/intangible/flock/flockmind/F = holder.owner
+	for(var/datum/unlockable_flock_structure/ufs as anything in F.flock.unlockableStructures)
+		if(ufs.check_unlocked())
+			var/obj/flock_structure/sT = ufs.structType //this is a gross hack, but needed for resolving flock_id
+			friendlyNames += initial(sT.flock_id) //flock_id is the friendly name for the structure
```

Comment:
Consider adding the 'friendly name' to the datum as a var. It can either get the initial flock_id from the structType it was initialized with or flock_id could be moved completely to the datum, though this would be troublesome for the inspect messages.